### PR TITLE
1701 obw   komponenten zur darstellung einer zusammenfassung der entscheidungen in stapel c

### LINKED ISSUE #1701: OBW - Komponenten zur Darstellung einer Zusammenfassung der Entscheidungen in Stapel C
# Beschreibung

Als Wahlvorstand möchte leichter erkennen können welche Beschlüsse insgesamt zu den bedenklichen Stimmzetteln gefasst wurden

# Voraussetzungen
- #1686
  - Die Wahlvorschläge sind in einem Store vorhanden
- #1687
  - Die Ergebnisse des Stapel C sind vorhanden

# Umsetzungshinweise

- liegen keine Ergebnisse vor ist der Tabelleninhalt leer und die Summe zeigt `0` an

## UI

<details>
<summary>Design Altanwendung</summary>

Image

</details>

# Akzeptanzkriterien

- [ ] es ist erkennbar wie oft bei bedenklichen Wahlbriefen für ungültig entschieden wurde
- [ ] es ist erkennbar wie oft für welchen Wahlvorschlag entschieden wurde
- [ ] Wahlvorschläge, für die nicht entschieden wurde, tauchen nicht auf
- [ ] es ist erkennbar wie viele Entscheidungen insgesamt getroffen wurden
- [ ] zuerst kommt ungültig dann die Wahlvorschläge entsprechend ihrer Ordnungszahl

### PR DESCRIPTION
**Pull request by MrSebastian**
# Beschreibung:

- eine einfache Komponente zur Zusammenfassung
- Daten kommen via Props rein

## Vorgänger
- #1843 ... Logik zur Ermittlung des Namens des ersten Kandidaten wird verwendet

### Frontend
- [x] [Naming Conventions][naming-conventions-link] beachtet
- [x] Doku aktualisiert - sehe ich erst mit der Integration wenn dem Nutzer die Funktion zur Verfügung gestellt wird
  - [Fachliche Beschreibung][fachliche-beschreibung-link]
- [X] ~~Unit-Tests gepflegt~~ - keine neue Logik
- [X] Texte auf Rechtschreibung und Grammatik geprüft

# Referenzen[^1]:

Closes #1701

> [^1]: _Nicht zutreffende Referenzen vor dem Speichern entfernen_

[naming-conventions-link]: https://it-at-m.github.io/Wahllokalsystem/technik/naming_conventions/
[fachliche-beschreibung-link]: https://it-at-m.github.io/Wahllokalsystem/about/#fachliche-anforderungen

## Summary by CodeRabbit

- New Features
  - Added a summary card for “Stapel C” that displays counts per proposal, includes an optional “Ungültig” row, and shows an overall total.
  - Automatically aggregates and shows only proposals with counts above zero.

- Documentation
  - Added Storybook examples covering default and edge cases (with/without valid and invalid results).

- Tests
  - Introduced comprehensive tests for result aggregation and totals, including edge cases (empty inputs, null values, and missing identifiers).

### CODE REVIEW

File: wls-gui-wahllokalsystem/stories/components/ergebnisermittlung/OBW/stapelC/TheOBWStapelCSummaryCard.stories.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        .kurzname("Wahlvorschlag 2")
+        .kandidaten(new Set([prepareKandidat().name("Kandidat 21").build()]))
+        .build(),
+      prepareWahlvorschlag()
+        .identifikator("id3")
+        .kurzname("Wahlvorschlag 3")
+        .kandidaten(new Set([prepareKandidat().name("Kandidat 31").build()]))
+        .build(),
+    ],
+  },
```
[the remark this answers is not in the document]
Comment (MrSebastian) [reply wls-gui-wahllokalsystem/stories/components/ergebnisermittlung/OBW/stapelC/TheOBWStapelCSummaryCard.stories.ts:78]: these props are optional and have default value, so no change.

File: wls-gui-wahllokalsystem/stories/components/ergebnismeldung/OBW/stapelA/TheOBWStapelACard.stories.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    wahlID,
+  },
+  decorators: [
+    (story) => {
+      return {
+        component: { story },
+        template: "<story />",
+      };
+    },
+  ],
```
[the remark this answers is not in the document]
Comment (MrSebastian) [reply wls-gui-wahllokalsystem/stories/components/ergebnismeldung/OBW/stapelA/TheOBWStapelACard.stories.ts]: is sth for #1843. If its right, it will got fixed there.

File: wls-gui-wahllokalsystem/stories/components/ergebnismeldung/OBW/stapelA/BaseRowOWBStapelA.stories.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      ]),
+    },
+  },
+  decorators: [
+    (story) => {
+      return {
+        component: { story },
+        template: "<table style='width:100%'><story /></table>",
+      };
+    },
```
[the remark this answers is not in the document]
Comment (MrSebastian) [reply wls-gui-wahllokalsystem/stories/components/ergebnismeldung/OBW/stapelA/BaseRowOWBStapelA.stories.ts]: is sth for #1843. If its right, it will got fixed there.

File: wls-gui-wahllokalsystem/stories/components/ergebnisermittlung/OBW/stapelC/TheOBWStapelCSummaryCard.stories.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+          wahlbezirkID: "wahlbezirkID",
+          stapelArt: StapelArtEnum.ObwCUngueltig,
+        })
+        .ergebnisse([
+          prepareErgebnis().ergebnis(1).build(),
+          prepareErgebnis().ergebnis(1).build(),
+        ])
+        .build(),
+    ];
+  },
```
[the remark this answers is not in the document]
Comment (MrSebastian) [reply wls-gui-wahllokalsystem/stories/components/ergebnisermittlung/OBW/stapelC/TheOBWStapelCSummaryCard.stories.ts]: done by 6a89dae19d3e04a7c31d0a69d19349091510f391

File: wls-gui-wahllokalsystem/src/components/ergebnisermittlung/OBW/stapelC/TheOBWStapelCSummaryCard.vue [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      </v-table>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import type { Ergebnis } from "@/types/ergebnismeldung/Ergebnis.ts";
+import type { Wahlvorschlag } from "@/types/wahlvorschlaege/Wahlvorschlag.ts";
+
+import { computed } from "vue";
```
[the remark this answers is not in the document]
Comment (MrSebastian) [reply wls-gui-wahllokalsystem/src/components/ergebnisermittlung/OBW/stapelC/TheOBWStapelCSummaryCard.vue:46]: fixed with 7bb347c152469925a87e510f20a0304433e59694

**Review by ThoGro — COMMENTED** [review]
Der Stapel C Teil passt 👍

**Review by ThoGro — DISMISSED** [review]

**Comment by MrSebastian** [comment]
@coderabbitai full review

**Comment by MrSebastian** [comment]
@ThoGro Ich habe den Code aus der Komponente in ein Composable ausgelagert, wie ich es auch bei Stapel A gemacht habe. Kannst du bitte erneut ein Review machen?

**Review by ThoGro — APPROVED** [review]
